server/neptune/workflows/internal/deploy/revision/queue/queue.go

```diff
@@ -19,28 +25,39 @@ type Deploy struct {
 	queue *priority
 
 	// mutable: default is unlocked
-	lock LockStatus
+	lock               LockState
+	lockStatusCallback func(workflow.Context, *Deploy)
 }
 
-func NewQueue() *Deploy {
+func NewQueue(callback func(workflow.Context, *Deploy)) *Deploy {
 	return &Deploy{
-		queue: newPriorityQueue(),
+		queue:              newPriorityQueue(),
+		lockStatusCallback: callback,
 	}
 
 }
 
-func (q *Deploy) SetLockStatusForMergedTrigger(status LockStatus) {
-	q.lock = status
+func (q *Deploy) GetLockState() LockState {
+	return q.lock
+}
+
+func (q *Deploy) SetLockForMergedQueue(ctx workflow.Context, state LockState) {
+	q.lock = state
+	q.lockStatusCallback(ctx, q)
 }
 
 func (q *Deploy) CanPop() bool {
-	return q.queue.HasItemsOfPriority(High) || (q.lock == UnlockedStatus && !q.queue.IsEmpty())
+	return q.queue.HasItemsOfPriority(High) || (q.lock.Status == UnlockedStatus && !q.queue.IsEmpty())
 }
 
 func (q *Deploy) Pop() (terraform.DeploymentInfo, error) {
 	return q.queue.Pop()
 }
 
+func (q *Deploy) GetMergedQueue() []terraform.DeploymentInfo {
```

Comment:
nit: `GetMergedElementsFromQueue`?
there might be better names, but when I read it, I saw it as a separate queue type that only had merged items

Reply:
hmm yeah i can do better probably